### PR TITLE
Feat [CHK-3619] Update mock api to support the new closePaymentResultError object

### PR DESCRIPTION
**Pull request by EmanueleBVtech**
depends on https://github.com/pagopa/pagopa-infra/pull/2644

#### List of Changes

Added support on mock api GET transactions for the new closePaymentResultError object

#### Motivation and Context

This is part of a bigger effort aiming to have a proper mapping for closePayment errors in the whole platform

#### How Has This Been Tested?

Manual testing
#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
...
     "generate:ts-from-yaml:api": "shx mkdir -p src/generated/payment_manager && shx mkdir -p src/generated/pagopa_proxy && rimraf src/generated/payment_manager/* && rimraf src/generated/pagopa_proxy/* && gen-api-models --strict 0 --api-spec api_payment_manager.yaml --out-dir src/generated/payment_manager --request-types && gen-api-models --strict 0 --api-spec https://raw.githubusercontent.com/pagopa/io-pagopa-proxy/v1.2.0/api-spec/api-pagopa-proxy.yaml --out-dir src/generated/pagopa_proxy --request-types",
-    "generate:ts-from-yaml:ecommerce": "shx rm -rf src/generated/ecommerce && shx mkdir -p src/generated/ecommerce && gen-api-models --api-spec https://raw.githubusercontent.com/pagopa/pagopa-infra/v1.256.0/src/domains/ecommerce-app/api/ecommerce-checkout/v1/_openapi.json.tpl --no-strict --out-dir src/generated/ecommerce",
+    "generate:ts-from-yaml:ecommerce": "shx rm -rf src/generated/ecommerce && shx mkdir -p src/generated/ecommerce && gen-api-models --api-spec https://raw.githubusercontent.com/pagopa/pagopa-infra/CHK-3613-update-get-transaction-definition/src/domains/ecommerce-app/api/ecommerce-checkout/v1/_openapi.json.tpl --no-strict --out-dir src/generated/ecommerce",
```
Comment (EmanueleBVtech) [inline package.json:23]: revert CHK-3613-update-get-transaction-definition after infra PR is merged

File: src/handlers/ecommerce/transaction.ts [hunk 2]
```diff
@@ -1432,148 +1478,232 @@ export const ecommerceGetTransaction: RequestHandler = async (req, res) => {
...
+          TransactionStatusEnum.CLOSURE_ERROR,
+          {
+            closePaymentResultError: {
+              description: "Node did not receive RPT yet",
+              statusCode: 422
+            },
+            sendPaymentResultOutcome: undefined
+          },
+          {
+            authorizationStatus: NpgAuthorizationStatus.DECLINED,
```
Comment (pietro-tota) [inline src/handlers/ecommerce/transaction.ts:1648]:
```suggestion
            authorizationStatus: NpgAuthorizationStatus.EXECUTED,
```
See comment on ecommerce-fe and checkout prs

Comment (EmanueleBVtech) [reply src/handlers/ecommerce/transaction.ts]: done here 7a15fede665b37e769eac0fdde94a02f8b0e8f91

**Review by pietro-tota — APPROVED** [review]
🚀

**Pull request merged**